### PR TITLE
fix(tooltip): always append to body and disappear on scroll

### PR DESCRIPTION
**Pull request by EladBezalel**
- Default animation origin has been set to bottom that mean the origin is `top center`

@jelbourn please review

### CODE REVIEW

File: src/components/tooltip/tooltip.js [hunk 1]
```diff
@@ -145,11 +146,17 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
       };
       var elementFocusedOnWindowBlur = false;
 
+      function windowScrollHandler() {
+        setVisible(false);
+      }
+      
       ngWindow.on('blur', windowBlurHandler);
       ngWindow.on('resize', debouncedOnResize);
+      document.addEventListener('scroll', windowScrollHandler, true);
```
Comment (jelbourn) [inline src/components/tooltip/tooltip.js]: Did you test this on Firefox? Scrolling behavior is slightly different there.

**Comment by kseamon** [comment]
http://www.quirksmode.org/dom/events/scroll.html

Try listening on window?

**Comment by kseamon** [comment]
Hacked together a workaround: https://github.com/angular/material/compare/master...kseamon:pr/6140?expand=1

**Comment by jelbourn** [comment]
Workaround seems like it would be good. @EladBezalel are you going to apply it and add a unit test?

**Comment by kseamon** [comment]
An explanation for the workaround (which should probably be added in a comment):

It seems like appending and/or repositioning the menu was triggering the scroll events (In FF only, which seems like buggy behavior but we're stuck with it). These events happen more or less right away, so nextTick fires after them.

**Comment by EladBezalel** [comment]
This is not working entirely, when checking the checkbox it's still disappearing.

**Comment by kseamon** [comment]
Well, it's another step towards hacky, but a timeout of 100ms instead of nextTick does the trick for the checkbox. Perhaps there's a longer animation going on when it triggers from the checkbox?

**Comment by jelbourn** [comment]
What do you mean by checking the checkbox?

**Comment by kseamon** [comment]
Better solution: remove transform: translate3d(0,0,0); from tooltip.scss. That seems to be the source of our FF misbehavior.

**Comment by jelbourn** [comment]
@kseamon that sounds like the best

**Comment by EladBezalel** [comment]
@jelbourn I'm on it

**Comment by kseamon** [comment]
I have not dived in with the chrome tools to look at paint performance to see if this makes any difference, but adding "will-change: opacity, height, width;" does not trigger the misbehavior but probably replaces any performance gains that the transform was accomplishing.

File: src/components/tooltip/tooltip.js (hunk 1 above)
Comment (michahell) [inline src/components/tooltip/tooltip.js]: This is probably causing a bug on firefox (at least in 1.09) where tooltips flash and immediately get removed...

**Pull request closed without merging**